src/native/eventpipe/ds-process-protocol.c

```diff
@@ -380,6 +521,74 @@ process_protocol_helper_get_process_info (
 	ep_exit_error_handler ();
 }
 
+static
+bool
+process_protocol_helper_get_process_info_2 (
+	DiagnosticsIpcMessage *message,
+	DiagnosticsIpcStream *stream)
+{
+	EP_ASSERT (message != NULL);
+	EP_ASSERT (stream != NULL);
+
+	bool result = false;
+	ep_char16_t *command_line = NULL;
+	ep_char16_t *os_info = NULL;
+	ep_char16_t *arch_info = NULL;
+	const ep_char16_t *managed_entrypoint_assembly_path_ref = ep_rt_entrypoint_assembly_path_get_utf16 ();
```

Comment:
Drop _ref and would be nice if we could handle this in same way as command_line, os_info, arch_info giving us one pattern to work with these strings.